File: src/path/InitialGuessPoint.cpp
// Copyright (c) TrajoptLib contributors

#include "path/InitialGuessPoint.h"

#include <iostream>

namespace trajopt {

InitialGuessPoint::InitialGuessPoint(double x, double y, double heading)
    : x(x), y(y), heading(heading) {}

std::ostream& operator<<(std::ostream& stream,
                         const InitialGuessPoint& guessPoint) {
  return stream << "{\"x\": " << guessPoint.x << ", \"y\": " << guessPoint.y
                << ", \"heading\": " << guessPoint.heading << "}";
}
}  // namespace trajopt

File: java/src/main/java/org/sleipnirgroup/trajoptlib/InitialGuessPoint.java
// Copyright (c) TrajoptLib contributors

package org.sleipnirgroup.trajoptlib;

public final class InitialGuessPoint {
    public final double x;
    public final double y;
    public final double heading;

    public InitialGuessPoint(double x, double y, double heading) {
        this.x = x;
        this.y = y;
        this.heading = heading;
    }

    @Override
    public String toString() {
        return "{\"x\": " + x
            + ", \"y\": " + y
            + ", \"heading\": " + heading
            + "}";
    }
}
